api/server/server.go

```diff
@@ -115,10 +117,12 @@ func parseMultipartForm(r *http.Request) error {
 }
 
 func httpError(w http.ResponseWriter, err error) {
+	if err == nil {
+		return
+	}
+
 	statusCode := http.StatusInternalServerError
-	// FIXME: this is brittle and should not be necessary.
-	// If we need to differentiate between different possible error types, we should
-	// create appropriate error types with clearly defined meaning.
+
```

Comment:
Wondering; shouldn't the actual brittle code below be removed? If I understand correctly, `Derror.Status()` always contains a status-code and a generic status code is set in line 124 above?